.github/workflows/build_and_test.yml

```diff
@@ -470,14 +474,20 @@ jobs:
             })
 
   trigger-e2e-tests:
-    # Depends on jobs that can get skipped
+    # !failure() && !cancelled() because it depends on jobs that can get skipped
     if: >-
       ${{
         (
-          !github.event.pull_request.draft
-          || contains( github.event.pull_request.labels.*.name, 'run-e2e-tests-in-draft')
-          || needs.meta.outputs.run-kind == 'push-main'
-        ) && !failure() && !cancelled()
+          (
+            needs.meta.outputs.run-kind == 'pr'
+            && (
+              !github.event.pull_request.draft
+              || contains(github.event.pull_request.labels.*.name, 'run-e2e-tests-in-draft')
+            )
+          )
+          || contains(fromJSON('["push-main", "storage-rc-pr", "proxy-rc-pr", "compute-rc-pr"]'), needs.meta.outputs.run-kind)
+        )
+        && !failure() && !cancelled()
```

Comment:
This previously had unintended behaviour:

```
!github.event.pull_request.draft || contains( github.event.pull_request.labels.*.name, 'run-e2e-tests-in-draft')
```

The first bit matches everything that is not a draft PR (including branch pushes), the second part broadens it to those draft PRs that have the label `run-e2e-tests-in-draft` set. The condition with `run-kind=push-main` didn't matter, because pushed branches always already were true in `!github.event.pull_request.draft`.

In addition to fixing that problem, we're now also *intentionally* running e2e tests on RC PRs, because of hotfixes. We could extend this to automatically check if we're in a hotfix and only run tests then if we want that, but I'd keep this as is for now.